pkg/hooks/listener.go

```diff
@@ -0,0 +1,182 @@
+			// if we find a pod at /hooks/before_install/usercreate, then the
+			// event is called "before_install"
+			event, err := l.determineEvent(result.Path)
+			if err != nil {
+				sub.WithField("err", err).Errorln("Couldn't determine hook path")
+				break
+			}
+
+			hookPod := pods.NewPod(result.Manifest.ID(), path.Join(l.DestinationDir, event, result.Manifest.ID()))
+			// install hooks as current user, not the pod ID. TODO: authorization checks for the pod
+			// to ensure they are permitted.
+			curUser, err := user.Current()
+			if err != nil {
+				sub.WithField("err", err).Errorln("could not get current user")
+			}
+			hookPod.RunAs = curUser.Username
+
+			// Figure out if we even need to install anything.
+			// Hooks aren't running services and so there isn't a need
+			// to write the current manifest to the reality store. Instead
+			// we just compare to the manifest on disk.
+			current, err := hookPod.CurrentManifest()
+			if err != nil && err != pods.NoCurrentManifest {
+				l.Logger.WithField("err", err).Errorln("Could not check current manifest")
+				errCh <- err
+				break
+			}
+
+			if err != pods.NoCurrentManifest && current.ID() == hookPod.Id {
+				// we are up-to-date, continue
+				break
+			}
+
+			// The manifest is new, go ahead and install
+			err = hookPod.Install(&result.Manifest)
+			if err != nil {
+				sub.WithField("err", err).Errorln("Could not install hook")
+				errCh <- err
+				break
+			}
+
+			_, err = hookPod.WriteCurrentManifest(&result.Manifest)
+			if err != nil {
+				sub.WithField("err", err).Errorln("Could not write current manifest")
+				errCh <- err
+				break
+			}
+
+			// Now that the pod is installed, link it up to the exec dir.
+			err = l.writeHook(event, hookPod, &result.Manifest)
+			if err != nil {
+				sub.WithField("err", err).Errorln("Could not write hook link")
+			}
+		}
+	}
+}
+
+func (l *Listener) determineEvent(pathInIntent string) (string, error) {
+	// The structure of a path in the hooks that we'll
+	// accept from consul is {prefix}/{event}/{podID}
+	split := strings.Split(pathInIntent, l.HookPrefix)
+	if len(split) == 0 {
+		return "", util.Errorf("The intent path %s didn't start with %s as expected", pathInIntent, l.HookPrefix)
+	}
+	tail := strings.Join(split[1:], "")
+	matches := eventPrefix.FindStringSubmatch(tail)
+	if len(matches) == 0 {
+		return "", util.Errorf("%s did not match regex %s", tail, eventPrefix.String())
+	}
+	return matches[1], nil
+}
+
+func (l *Listener) writeHook(event string, hookPod *pods.Pod, manifest *pods.PodManifest) error {
+	eventExecDir := path.Join(l.ExecDir, event)
+	err := os.MkdirAll(eventExecDir, 0755)
+	if err != nil {
+		return util.Errorf("Couldn't make event exec dir %s", eventExecDir)
+	}
+	launchables, err := hookPod.GetLaunchables(manifest)
+	if err != nil {
+		return err
+	}
+
+	// First remove any pre-existing hooks for that pod.
+	podHookPattern := path.Join(eventExecDir, fmt.Sprintf("%s__*", hookPod.Id))
```

Comment:
Here I'm kind of nastily globbing every symlink from the previous version of the pod and removing them. Notice two bad things - one that there isn't a better grouping mechanism (like a sub folder) for hooks created by a particular pod - and two, that I just happen to know that `{podId}__*` is the wildcard for the name of hooks assigned to a pod. 

To do: refactor the hooks library to execute hooks in directories recursively.